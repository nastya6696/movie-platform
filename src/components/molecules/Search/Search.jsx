import styles from './styles.module.scss';
import {Button} from "../../atoms";
import {useState} from "react";

export const Search = () => {
  const [searchValue, setSearchValue] = useState('');

  const handleChange = (e) => {
    setSearchValue(e.target.value);
  }

  return (
    <>
      <input
        type="text"
        className={styles.searchBar}
        placeholder="What do you want to watch?"
        name="search"
        value={searchValue}
        onChange={handleChange}
      />
      <Button name="SEARCH"/>
    </>
  );
};
