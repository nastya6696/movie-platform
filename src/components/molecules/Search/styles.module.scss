@import "src/assets/scss/common";

.searchBar {
  width: 57%;
  height: 45px;
  border-radius: 4px;
  border: none;
  background-color: $grey-background-20-transparent;
  @include fontSizes(20px, 25px);
  font-weight: 100;
  margin-right: 15px;
}
