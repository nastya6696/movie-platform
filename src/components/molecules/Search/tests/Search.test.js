import React from 'react';
import {Search} from "../Search";
import {create} from 'react-test-renderer';

jest.mock('../../../atoms/Button', () => ({
  Button: (({name, handleClick}) => `
     (--- Button ---): props {
      name: ${name},
      handleClick: ${handleClick}
     }`
  )
}));

describe('Search', () => {
  it('should create simple Search component', () => {
    const setSearchValue = jest.fn();
    jest.spyOn(React, 'useState')
      .mockImplementationOnce(() => ['', setSearchValue]);

    const component = create(<Search />);

    expect(component).toMatchSnapshot();
  })
});
