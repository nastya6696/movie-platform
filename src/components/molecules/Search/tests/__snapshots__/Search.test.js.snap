// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should create simple Search component 1`] = `
Array [
  <input
    className="searchBar"
    name="search"
    onChange={[Function]}
    placeholder="What do you want to watch?"
    type="text"
    value=""
  />,
  "
     (--- Button ---): props {
      name: SEARCH,
      handleClick: undefined
     }",
]
`;
